File: src/gui/Menu.lua
--- src/gui/Menu.lua
-- @class Menu
-- Handles the game menu overlay

local Menu = {}

function Menu:load()
    self.active = false

    -- Set up menu appearance
    self.width = 400
    self.height = 300
    self.x = (love.graphics.getWidth() - self.width) / 2
    self.y = (love.graphics.getHeight() - self.height) / 2

    -- Background color
    self.bgColor = {
        0.2,
        0.2,
        0.2,
        0.9,
    }

    -- Button properties
    self.buttonHeight = 60
    self.buttonWidth = 300
    self.buttonMargin = 20

    -- Create buttons
    self.buttons = {
        {
            text = "Resume",
            x = self.x + (self.width - self.buttonWidth) / 2,
            y = self.y + 50,
            width = self.buttonWidth,
            height = self.buttonHeight,
            action = function()
                self:toggle()
            end,
        },
        {
            text = "Settings",
            x = self.x + (self.width - self.buttonWidth) / 2,
            y = self.y + 50 + self.buttonHeight + self.buttonMargin,
            width = self.buttonWidth,
            height = self.buttonHeight,
            action = function()
                print("Settings menu (not implemented)")
            end,
        },
        {
            text = "Quit",
            x = self.x + (self.width - self.buttonWidth) / 2,
            y = self.y + 50 + (self.buttonHeight + self.buttonMargin) * 2,
            width = self.buttonWidth,
            height = self.buttonHeight,
            action = function()
                love.event.quit()
            end,
        },
    }

    -- Font for the menu
    self.font = love.graphics.newFont("assets/fonts/public-pixel-font.ttf", 24)
end

function Menu:update(dt)
    if not self.active then
        return
    end

    -- Get mouse position
    local mx, my = love.mouse.getPosition()

    -- Check button hover state
    for i, button in ipairs(self.buttons) do
        button.hover = mx >= button.x and mx <= button.x + button.width and my >= button.y and my <= button.y + button.height
    end
end

function Menu:draw()
    if not self.active then
        return
    end

    -- Store original font to restore later
    local originalFont = love.graphics.getFont()
    love.graphics.setFont(self.font)

    -- Draw semi-transparent background
    love.graphics.setColor(self.bgColor)
    love.graphics.rectangle("fill", self.x, self.y, self.width, self.height)

    -- Draw border
    love.graphics.setColor(1, 1, 1, 1)
    love.graphics.rectangle("line", self.x, self.y, self.width, self.height)

    -- Draw title
    love.graphics.setColor(1, 1, 1, 1)
    local title = "Menu"
    local titleWidth = self.font:getWidth(title)
    love.graphics.print(title, self.x + (self.width - titleWidth) / 2, self.y + 10)

    -- Draw buttons
    for i, button in ipairs(self.buttons) do
        -- Button background
        if button.hover then
            love.graphics.setColor(0.4, 0.4, 0.4, 1)
        else
            love.graphics.setColor(0.3, 0.3, 0.3, 1)
        end

        love.graphics.rectangle("fill", button.x, button.y, button.width, button.height)

        -- Button border
        love.graphics.setColor(1, 1, 1, 1)
        love.graphics.rectangle("line", button.x, button.y, button.width, button.height)

        -- Button text
        love.graphics.setColor(1, 1, 1, 1)
        local textWidth = self.font:getWidth(button.text)
        local textHeight = self.font:getHeight()
        love.graphics.print(button.text, button.x + (button.width - textWidth) / 2, button.y + (button.height - textHeight) / 2)
    end

    -- Restore original font
    love.graphics.setFont(originalFont)
end

function Menu:toggle()
    self.active = not self.active
end

function Menu:mousepressed(x, y, button)
    if not self.active or button ~= 1 then
        return
    end

    -- Check if any buttons were clicked
    for i, btn in ipairs(self.buttons) do
        if x >= btn.x and x <= btn.x + btn.width and y >= btn.y and y <= btn.y + btn.height then
            btn.action()
            return
        end
    end
end

return Menu

File: internal/gui/menu.go
package gui

import (
	"image/color"
	"log"

	"github.com/hajimehoshi/ebiten/v2"
	"github.com/hajimehoshi/ebiten/v2/text/v2"
	"github.com/hajimehoshi/ebiten/v2/vector"
)

// Menu constants — scaled for 640×360 internal resolution.
const (
	menuWidth        = 200.0
	menuHeight       = 150.0
	menuButtonWidth  = 150.0
	menuButtonHeight = 28.0
	menuButtonMargin = 10.0
	menuFontSize     = 10.0
)

// Button is a clickable menu button.
type Button struct {
	Text   string
	X, Y   float64
	W, H   float64
	Hover  bool
	Action func() error
}

// Menu is a pause menu overlay with buttons.
type Menu struct {
	Active  bool
	buttons []*Button
	font    *text.GoTextFace
	x, y    float64
}

// NewMenu creates a centered pause menu.
func NewMenu(screenW, screenH float64, toggleFn func()) (*Menu, error) {
	fontSrc, err := LoadFont("assets/fonts/public-pixel-font.ttf")
	if err != nil {
		return nil, err
	}

	face := &text.GoTextFace{
		Source: fontSrc,
		Size:   menuFontSize,
	}

	mx := (screenW - menuWidth) / 2
	my := (screenH - menuHeight) / 2
	btnX := mx + (menuWidth-menuButtonWidth)/2

	m := &Menu{
		font: face,
		x:    mx,
		y:    my,
	}

	m.buttons = []*Button{
		{
			Text:   "Resume",
			X:      btnX,
			Y:      my + 25,
			W:      menuButtonWidth,
			H:      menuButtonHeight,
			Action: func() error { toggleFn(); return nil },
		},
		{
			Text:   "Settings",
			X:      btnX,
			Y:      my + 25 + menuButtonHeight + menuButtonMargin,
			W:      menuButtonWidth,
			H:      menuButtonHeight,
			Action: func() error { log.Println("Settings not implemented"); return nil },
		},
		{
			Text:   "Quit",
			X:      btnX,
			Y:      my + 25 + (menuButtonHeight+menuButtonMargin)*2,
			W:      menuButtonWidth,
			H:      menuButtonHeight,
			Action: func() error { return ebiten.Termination },
		},
	}

	return m, nil
}

// Toggle flips the menu active state.
func (m *Menu) Toggle() {
	m.Active = !m.Active
}

// Update checks mouse hover on buttons.
func (m *Menu) Update() {
	if !m.Active {
		return
	}

	mx, my := ebiten.CursorPosition()
	fx, fy := float64(mx), float64(my)

	for _, b := range m.buttons {
		b.Hover = fx >= b.X && fx <= b.X+b.W && fy >= b.Y && fy <= b.Y+b.H
	}
}

// Draw renders the pause menu overlay.
func (m *Menu) Draw(screen *ebiten.Image) {
	if !m.Active {
		return
	}

	// Semi-transparent background
	vector.DrawFilledRect(screen, float32(m.x), float32(m.y), menuWidth, menuHeight, color.NRGBA{51, 51, 51, 230}, true)

	// Border
	vector.StrokeRect(screen, float32(m.x), float32(m.y), menuWidth, menuHeight, 1, color.White, true)

	// Title
	title := "Menu"
	tw, _ := MeasureText(title, m.font)
	DrawText(screen, title, m.x+(menuWidth-tw)/2, m.y+10, m.font, color.White)

	// Buttons
	for _, b := range m.buttons {
		// Button background
		bgClr := color.NRGBA{77, 77, 77, 255}
		if b.Hover {
			bgClr = color.NRGBA{102, 102, 102, 255}
		}
		vector.DrawFilledRect(screen, float32(b.X), float32(b.Y), float32(b.W), float32(b.H), bgClr, true)

		// Button border
		vector.StrokeRect(screen, float32(b.X), float32(b.Y), float32(b.W), float32(b.H), 1, color.White, true)

		// Button text (centered)
		tw, th := MeasureText(b.Text, m.font)
		DrawText(screen, b.Text, b.X+(b.W-tw)/2, b.Y+(b.H-th)/2, m.font, color.White)
	}
}

// MousePressed checks if a button was clicked and fires its action.
func (m *Menu) MousePressed(x, y int) error {
	if !m.Active {
		return nil
	}

	fx, fy := float64(x), float64(y)
	for _, b := range m.buttons {
		if fx >= b.X && fx <= b.X+b.W && fy >= b.Y && fy <= b.Y+b.H {
			return b.Action()
		}
	}
	return nil
}
